### PR TITLE
Add snapshot parameter flag

### LINKED ISSUE #235: Add snapshot parameters file flag
When creating snapshot,  VolumeSnapshotClass's parameters will passed through CreateSnapshotRequest.parameters. We could add a flag to pass VolumeSnapshotClass parameters
 in sanity test just like StorageClass parameters.

### PR DESCRIPTION
**Pull request by wnxn**
Signed-off-by: Xin Wang <wileywang@yunify.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
/kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #235 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Add '--csi.testsnapshotparameters' flag
```

### CODE REVIEW

File: pkg/sanity/sanity.go [hunk 1]
```diff
@@ -163,18 +165,30 @@ func newContext(reqConfig *Config) *SanityContext {
...
+			panic(fmt.Sprintf("failed to read file %q: %v", volParamPath, err))
 		}
 		err = yaml.Unmarshal(yamlFile, &reqConfig.TestVolumeParameters)
 		if err != nil {
 			panic(fmt.Sprintf("error unmarshaling yaml: %v", err))
 		}
 	}
 
+	snapParamPath := reqConfig.TestSnapshotParametersFile
+	if len(snapParamPath) != 0 {
```
Comment (pohly) [inline pkg/sanity/sanity.go]: Instead copying the unmarshaling code from the `reqConfig.TestVolumeParametersFile` handling above, can you move it into a helper function?

File: pkg/sanity/sanity.go [hunk 2]
```diff
@@ -69,10 +69,12 @@ type Config struct {
...
 
 	// Target size for ExpandVolume requests. If not specified it defaults to TestVolumeSize + 1 GB
-	TestVolumeExpandSize      int64
-	TestVolumeParametersFile  string
-	TestVolumeParameters      map[string]string
-	TestNodeVolumeAttachLimit bool
+	TestVolumeExpandSize       int64
+	TestVolumeParametersFile   string
+	TestVolumeParameters       map[string]string
+	TestSnapshotParametersFile string
```
Comment (pohly) [inline pkg/sanity/sanity.go]: Can you add the new fields in a separate block (i.e. don't mix with `TestVolumeExpand*`) and add documentation? I know it's missing also for the other fields, but that's not a good example to follow :man_shrugging:

File: pkg/sanity/sanity.go (hunk 1 above)
Comment (wnxn) [reply pkg/sanity/sanity.go]: Thanks for your review. I will amend my commit.

File: pkg/sanity/sanity.go [hunk 3]
```diff
@@ -357,6 +354,22 @@ func loadSecrets(path string) (*CSISecrets, error) {
 	return &creds, nil
 }
 
+// loadMap reads a map from given file path.
+func loadMap(path string) map[string]string {
```
Comment (pohly) [inline pkg/sanity/sanity.go]: I think you can make this more generic:
```
func loadFromFile(from string, to interface{}) {
    if len(path) != 0 {
...
        err = yaml.Unmarshal(yamlFile, to)
    }
}
```

Then call it with:
```
loadFromFile(reqConfig.TestSnapshotParametersFile, &reqConfig.TestSnapshotParameters)
```

Untested, but I think it'll work, and then we have something that could also become useful for populating other data structures.

Comment (wnxn) [reply pkg/sanity/sanity.go]: OK, I will update and test it.

Comment (wnxn) [reply pkg/sanity/sanity.go]: Hi @pohly , The method is worked and I updated this PR. Would you please review again?

**Pull request merged**